libr/util/sys.c

```diff
@@ -363,26 +363,34 @@ R_API int r_sys_crash_handler(const char *cmd) {
 
 	free (crash_handler_cmd);
 	crash_handler_cmd = strdup (cmd);
-	sigact.sa_handler = signal_handler;
-	sigemptyset (&sigact.sa_mask);
-	sigact.sa_flags = 0;
-	sigaddset (&sigact.sa_mask, SIGINT);
-	sigaddset (&sigact.sa_mask, SIGSEGV);
-	sigaddset (&sigact.sa_mask, SIGBUS);
-	sigaddset (&sigact.sa_mask, SIGQUIT);
-	sigaddset (&sigact.sa_mask, SIGHUP);
+	sigemptyset (&mask);
+	sigaddset (&mask, SIGINT);
+	sigaddset (&mask, SIGSEGV);
+	sigaddset (&mask, SIGBUS);
+	sigaddset (&mask, SIGQUIT);
+	sigaddset (&mask, SIGHUP);
 
-	sigaction (SIGINT, &sigact, (struct sigaction *)NULL);
-	sigaction (SIGSEGV, &sigact, (struct sigaction *)NULL);
-	sigaction (SIGBUS, &sigact, (struct sigaction *)NULL);
-	sigaction (SIGQUIT, &sigact, (struct sigaction *)NULL);
-	sigaction (SIGHUP, &sigact, (struct sigaction *)NULL);
+	r_sys_sigaction (SIGINT, signal_handler, &mask, 0);
+	r_sys_sigaction (SIGSEGV, signal_handler, &mask, 0);
+	r_sys_sigaction (SIGBUS, signal_handler, &mask, 0);
+	r_sys_sigaction (SIGQUIT, signal_handler, &mask, 0);
+	r_sys_sigaction (SIGHUP, signal_handler, &mask, 0);
 	return true;
 #else
 	return false;
 #endif
 }
 
+#if __UNIX__
+R_API int r_sys_sigaction(int sig, void(*handler) (int), sigset_t *mask, int flags) {
```

Comment:
must be `R_API RSysSigAction r_sys_sigaction (int sig, ...);`

```
#define RSysSIgAction sigset_t
#else
#define RSysSIgAction int
```
make it vararg, to iterate over {sig, handler, flags} until -1 is found in the `sig`.

the sigset_t must be initialized here

Reply:
flags seems to be always 0, can we just hardcode this and reduce the arglist ?

Reply:
> flags seems to be always 0, can we just hardcode this and reduce the arglist ?

￼If you want, but that seems like a bad/not-future-proof API to me.